src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -60,8 +62,45 @@ def handleRequest(request)
         response["X-Puppet-Version"])
   end
 
-  def compileCatalog(_request_data)
-    {}
+  def convert_java_args_to_ruby(hash)
+    Hash[hash.collect do |key, value|
+        # Stolen and modified from params_to_ruby in handler.rb
+        newkey = key.to_s
+        newkey.slice!(0)
```

Comment:
wth

Reply:
Is to remove a leading colon?

Reply:
Yes. I wrote that method to unblock work for facts processing. It's not perfect!...or is it?

Reply:
Maybe it is...

![](https://media.giphy.com/media/zGgd5ZQWB13La/giphy.gif)